### PR TITLE
[@poll error] attribute

### PR DESCRIPTION
**Pull request by mshinwell**
This backports the support for the `[@poll error]` attribute from ocaml/ocaml#10462 and adds the necessary code to propagate such attribute through Flambda 2.

Based on #740.

### CODE REVIEW

File: ocaml/lambda/translattribute.ml [hunk 1]
```diff
@@ -342,7 +401,8 @@ let get_tailcall_attribute e =
 let check_attribute e {Parsetree.attr_name = { txt; loc }; _} =
   match txt with
   | "inline" | "ocaml.inline"
-  | "specialise" | "ocaml.specialise" -> begin
+  | "specialise" | "ocaml.specialise"
+  | "poll" -> begin
```
Comment (xclerc) [inline ocaml/lambda/translattribute.ml:406]: (Why not also `"ocaml.poll"`?)

**Review by xclerc — APPROVED** [review]
(As noted in a CR, a `Cfgize`-based pipeline would not honor the attribute.)

File: ocaml/lambda/translattribute.ml (hunk 1 above)
Comment (mshinwell) [reply ocaml/lambda/translattribute.ml:464]: Fixed

**Comment by mshinwell** [comment]
As a note, this PR doesn't import the safepoints changes to the `ocaml/` subdirectory except the few that are absolutely necessary.  A few poll-error tests have been moved into the toplevel `testsuite/` directory as a consequence; once `ocaml-jst` has been rebased to 4.14 these will be redundant (I have made a note to remove these at that time).  The changes to the `ocaml/` subdirectory should naturally come in via the 4.14 rebase, but we can make a separate PR if that proves problematic.

**Pull request merged**